docs/topics/kotlin-evolution-principles.md

```diff
@@ -0,0 +1,248 @@
+  _in preview_, it has a [stability level](components-stability.md#stability-levels-explained).
+
+* **Stable**. The language feature is now a first-class citizen in the Kotlin language.
+  We guarantee its backward compatibility and that we'll provide tooling support.
+
+* **Revoked**. We have revoked the proposal and will not implement the feature in the Kotlin language.
+  We may revoke a feature that is _in preview_ if it is not a good fit for Kotlin.
+
+[See the full list of Kotlin language proposals and their statuses](kotlin-language-features-and-proposals.md).
+
+### Status of different components
+
+Learn more about the [stability status of different components in Kotlin (such as Kotlin/JVM, JS, and Native compilers, various libraries)](components-stability.md)
+
+## Libraries
+
+A language is nothing without its ecosystem, so we pay extra attention to enabling smooth library evolution.
+
+Ideally, a new version of a library can be used as a "drop-in replacement" for an older version. 
+This means that upgrading a binary dependency should not break anything, 
+even if the application is not recompiled (this is possible under dynamic linking). 
+
+On the one hand, to achieve this, the compiler has to provide certain ABI stability guarantees under the constraints of separate compilation.
+This is why every change in the language is examined from a binary compatibility standpoint.
+
+On the other hand, a lot depends on the library authors being careful about which changes are safe to make. 
+Thus, it's crucial that library authors understand how source changes affect compatibility
+and follow certain best practices to keep both APIs and ABIs of their libraries stable.
+Here are some assumptions that we make when considering language changes from the library evolution standpoint:
+
+* Library code should always specify return types of public/protected functions and properties explicitly,
+    thus never relying on type inference for public API. Subtle changes in type inference may cause return types to change inadvertently,
+    leading to binary compatibility issues.
+* Overloaded functions and properties provided by the same library should do essentially the same thing. 
+    Changes in type inference may result in more precise static types to be known at call sites,
+    causing changes in overload resolution.
+
+Library authors can use the `@Deprecated` and [`@RequiresOptIn`](opt-in-requirements.md) annotations
+to control the evolution of their API surface. Note that `@Deprecated(level=HIDDEN)`
+can be used to preserve binary compatibility even for declarations removed from the API.
+
+Also, by convention, packages named "internal" are not considered public API. 
+All API residing in packages named "experimental" is considered pre-stable and can change at any moment.
+
+We evolve the Kotlin Standard Library (`kotlin-stdlib`) for stable platforms according to the principles stated above.
```

Comment:
```suggestion
We evolve the Kotlin standard library (`kotlin-stdlib`) for stable platforms according to the principles stated above.
```
Similar to previous comment about standard library being lower case.

Reply:
Rejected.